### PR TITLE
Feature/tri 1310 update edc to 0.4.0

### PR DESCRIPTION
**Pull request by ds-jhartmann**

### CODE REVIEW

File: irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcControlPlaneClient.java [hunk 1]
```diff
@@ -129,32 +142,49 @@ private CatalogRequest buildCatalogRequest(final int offset, final String provid
...
     }
 
-    /* package */ TransferProcessId startTransferProcess(final TransferProcessRequest request) {
-        return edcRestTemplate.exchange(config.getControlplane().getEndpoint().getData() + "/transferprocess",
-                HttpMethod.POST, new HttpEntity<>(request, headers()), TransferProcessId.class).getBody();
+    private NegotiationState getContractNegotiationState(final IdResponseDto negotiationId,
+            final HttpEntity<Object> objectHttpEntity) {
+        final String negotiationStateResponse = edcRestTemplate.exchange(
+                config.getControlplane().getEndpoint().getData() + "/contractnegotiations/" + negotiationId.getId()
+                        + "/state", HttpMethod.GET, objectHttpEntity, String.class).getBody();
```
Comment (ds-jkreutzfeld) [inline irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcControlPlaneClient.java:149]: Again, maybe that URL pattern should also be configurable?

File: irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcControlPlaneClient.java [hunk 2]
```diff
@@ -129,32 +142,49 @@ private CatalogRequest buildCatalogRequest(final int offset, final String provid
...
+                StandardCharsets.UTF_8);
     }
 
-    /* package */ CompletableFuture<TransferProcessResponse> getTransferProcess(
-            final TransferProcessId transferProcessId) {
+    private NegotiationResponse getContractNegotiationResponse(final IdResponseDto negotiationId,
+            final HttpEntity<Object> objectHttpEntity) {
+        final String negotiationResponse = edcRestTemplate.exchange(
+                config.getControlplane().getEndpoint().getData() + "/contractnegotiations/" + negotiationId.getId(),
+                HttpMethod.GET, objectHttpEntity, String.class).getBody();
```
Comment (ds-jkreutzfeld) [inline irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcControlPlaneClient.java:158]: see above

File: irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcControlPlaneClient.java [hunk 3]
```diff
@@ -129,32 +142,49 @@ private CatalogRequest buildCatalogRequest(final int offset, final String provid
...
+
+        return edcTransformer.transformJsonToNegotiationResponse(Objects.requireNonNull(negotiationResponse),
+                StandardCharsets.UTF_8);
+    }
+
+    /* package */ IdResponseDto startTransferProcess(final TransferProcessRequest request) {
+        final String jsonObject = edcTransformer.transformTransferProcessRequestToJson(request).toString();
+        return edcRestTemplate.exchange(config.getControlplane().getEndpoint().getData() + "/transferprocesses",
+                HttpMethod.POST, new HttpEntity<>(jsonObject, headers()), IdResponseDto.class).getBody();
```
Comment (ds-jkreutzfeld) [inline irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcControlPlaneClient.java:167]: see above

File: irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcControlPlaneClient.java [hunk 4]
```diff
@@ -169,9 +199,26 @@ private CatalogRequest buildCatalogRequest(final int offset, final String provid
 
     }
 
+    private NegotiationState getTransferProcessState(final IdResponseDto transferProcessId,
+            final HttpEntity<Object> objectHttpEntity) {
+        final String transferProcessStateResponse = edcRestTemplate.exchange(
+                config.getControlplane().getEndpoint().getData() + "/transferprocesses/" + transferProcessId.getId()
+                        + "/state", HttpMethod.GET, objectHttpEntity, String.class).getBody();
```
Comment (ds-jkreutzfeld) [inline irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcControlPlaneClient.java:206]: see above

File: irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcControlPlaneClient.java [hunk 5]
```diff
@@ -169,9 +199,26 @@ private CatalogRequest buildCatalogRequest(final int offset, final String provid
...
+        return edcTransformer.transformJsonToNegotiationState(Objects.requireNonNull(transferProcessStateResponse),
+                StandardCharsets.UTF_8);
+    }
+
+    private TransferProcessResponse getTransferProcessResponse(final IdResponseDto transferProcessId,
+            final HttpEntity<Object> objectHttpEntity) {
+        return edcRestTemplate.exchange(
+                config.getControlplane().getEndpoint().getData() + "/transferprocesses/" + transferProcessId.getId(),
+                HttpMethod.GET, objectHttpEntity, TransferProcessResponse.class).getBody();
```
Comment (ds-jkreutzfeld) [inline irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EdcControlPlaneClient.java:215]: see above

File: charts/edc-consumer/Chart.yaml [hunk 6]
```diff
@@ -14,13 +14,13 @@ type: application
...
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 1.1.2
+version: 1.1.3
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application. Versions are not expected to
 # follow Semantic Versioning. They should reflect the version the application is using.
 # It is recommended to use it with quotes.
-appVersion: "0.3.0"
+appVersion: "0.4.0"
```
[the remark this answers is not in the document]
Comment (ds-jhartmann) [reply charts/edc-consumer/Chart.yaml]: Latest version of the charts were in charts/edc-test. Moved it to charts/edc-consumer.

**Review by ds-jkreutzfeld — DISMISSED** [review]

**Review by ds-jkreutzfeld — APPROVED** [review]

**Pull request merged**